`pow` can be a variable?

``` r
library(nlmixr2)
#> Warning: package 'nlmixr2' was built under R version 4.2.2
#> Loading required package: nlmixr2data


one.compartment <- function() {
  ini({
    tka <- 0.45 # Log Ka
    tcl <- 1 # Log Cl
    tv <- 3.45    # Log V
    eta.ka ~ 0.6
    eta.cl ~ 0.3
    eta.v ~ 0.1
    add.sd <- 0.7
    pow <- fix(1)
  })
  # and a model block with the error sppecification and model specification
  model({
    ka <- exp(tka + eta.ka)
    cl <- exp(tcl + eta.cl)
    v <- exp(tv + eta.v)
    d/dt(depot) = -ka * depot
    d/dt(center) = ka * depot - cl / v * center
    cp = center / v
    cp ~ pow(add.sd, pow)
  })
}


fit <- nlmixr2(one.compartment, theo_sd,  est="focei", foceiControl(print=0))
#> ℹ parameter labels from comments will be replaced by 'label()'
#> → loading into symengine environment...
#> → pruning branches (`if`/`else`) of full model...
#> ✔ done
#> → calculate jacobian
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → calculate sensitivities
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → calculate ∂(f)/∂(η)
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → calculate ∂(R²)/∂(η)
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → finding duplicate expressions in inner model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → optimizing duplicate expressions in inner model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → finding duplicate expressions in EBE model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → optimizing duplicate expressions in EBE model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → compiling inner model...
#> ✔ done
#> → finding duplicate expressions in FD model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → optimizing duplicate expressions in FD model...
#> [====|====|====|====|====|====|====|====|====|====] 0:00:00
#> → compiling EBE model...
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> rx_90b68597570124745132ef0971cce571_x64.c: In function 'rx_90b68597570124745132ef0971cce571_x64_dydt':
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:55: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                       ^
#> rx_90b68597570124745132ef0971cce571_x64.c:600:20: note: in expansion of macro 'Rx_pow'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:51: error: called object 'pow' is not a function or function pointer
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                   ^~~
#> rx_90b68597570124745132ef0971cce571_x64.c:600:9: note: in expansion of macro 'Rx_pow_di'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |         ^~~~~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:72: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                                        ^
#> rx_90b68597570124745132ef0971cce571_x64.c:600:20: note: in expansion of macro 'Rx_pow'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> rx_90b68597570124745132ef0971cce571_x64.c: In function 'rx_90b68597570124745132ef0971cce571_x64_calc_lhs':
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:55: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                       ^
#> rx_90b68597570124745132ef0971cce571_x64.c:766:20: note: in expansion of macro 'Rx_pow'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:51: error: called object 'pow' is not a function or function pointer
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                   ^~~
#> rx_90b68597570124745132ef0971cce571_x64.c:766:9: note: in expansion of macro 'Rx_pow_di'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |         ^~~~~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:72: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                                        ^
#> rx_90b68597570124745132ef0971cce571_x64.c:766:20: note: in expansion of macro 'Rx_pow'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> make: *** [c:/R/R-42~1.1/etc/x64/Makeconf:253: rx_90b68597570124745132ef0971cce571_x64.o] Error 1
#> error building model
#> gcc  -I"c:/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"  -O3  -I"C:\R\R-42~1.1\library\rxode2\include" -I"C:/R/R-4.2.1/library/rxode2parse/include"   -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c rx_90b68597570124745132ef0971cce571_x64.c -o rx_90b68597570124745132ef0971cce571_x64.o
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> rx_90b68597570124745132ef0971cce571_x64.c: In function 'rx_90b68597570124745132ef0971cce571_x64_dydt':
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:55: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                       ^
#> rx_90b68597570124745132ef0971cce571_x64.c:600:20: note: in expansion of macro 'Rx_pow'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:51: error: called object 'pow' is not a function or function pointer
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                   ^~~
#> rx_90b68597570124745132ef0971cce571_x64.c:600:9: note: in expansion of macro 'Rx_pow_di'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |         ^~~~~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:72: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                                        ^
#> rx_90b68597570124745132ef0971cce571_x64.c:600:20: note: in expansion of macro 'Rx_pow'
#>   600 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:508:10: note: declared here
#>   508 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> rx_90b68597570124745132ef0971cce571_x64.c: In function 'rx_90b68597570124745132ef0971cce571_x64_calc_lhs':
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:55: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                       ^
#> rx_90b68597570124745132ef0971cce571_x64.c:766:20: note: in expansion of macro 'Rx_pow'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:51: error: called object 'pow' is not a function or function pointer
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                   ^~~
#> rx_90b68597570124745132ef0971cce571_x64.c:766:9: note: in expansion of macro 'Rx_pow_di'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |         ^~~~~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> In file included from rx_90b68597570124745132ef0971cce571_x64.c:115:
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:21:48: error: called object 'pow' is not a function or function pointer
#>    21 | #define Rx_pow(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow(a, b))
#>       |                                                ^~~
#> C:/R/R-4.2.1/library/rxode2parse/include/rxode2_model_shared.h:22:72: note: in definition of macro 'Rx_pow_di'
#>    22 | #define Rx_pow_di(a, b) (&_solveData->useStdPow ? pow(a, b) : R_pow_di(a, b))
#>       |                                                                        ^
#> rx_90b68597570124745132ef0971cce571_x64.c:766:20: note: in expansion of macro 'Rx_pow'
#>   766 |   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>       |                    ^~~~~~
#> rx_90b68597570124745132ef0971cce571_x64.c:671:10: note: declared here
#>   671 |   double pow;
#>       |          ^~~
#> make: *** [c:/R/R-42~1.1/etc/x64/Makeconf:253: rx_90b68597570124745132ef0971cce571_x64.o] Error 1
#> #define _getRxSolve_ _rxa414c27d93d5d5d5e72becc146c5f97b0
#> #define _solveData _rxa414c27d93d5d5d5e72becc146c5f97b1
#> #define _assign_ptr _rxa414c27d93d5d5d5e72becc146c5f97b2
#> #define _rxRmModelLib _rxa414c27d93d5d5d5e72becc146c5f97b3
#> #define _rxGetModelLib _rxa414c27d93d5d5d5e72becc146c5f97b4
#> #define _old_c _rxa414c27d93d5d5d5e72becc146c5f97b5
#> #define _ptrid _rxa414c27d93d5d5d5e72becc146c5f97b6
#> #define _rxIsCurrentC _rxa414c27d93d5d5d5e72becc146c5f97b7
#> #define _sumPS _rxa414c27d93d5d5d5e72becc146c5f97b8
#> #define _prodPS _rxa414c27d93d5d5d5e72becc146c5f97b9
#> #define _prodType _rxa414c27d93d5d5d5e72becc146c5f97b10
#> #define _sumType _rxa414c27d93d5d5d5e72becc146c5f97b11
#> #define _update_par_ptr _rxa414c27d93d5d5d5e72becc146c5f97b12
#> #define _getParCov _rxa414c27d93d5d5d5e72becc146c5f97b13
#> #define _rxode2_rxAssignPtr _rxa414c27d93d5d5d5e72becc146c5f97b14
#> #define _rxQr _rxa414c27d93d5d5d5e72becc146c5f97b15
#> #define _compareFactorVal _rxa414c27d93d5d5d5e72becc146c5f97b16
#> #define _sum _rxa414c27d93d5d5d5e72becc146c5f97b17
#> #define _sign _rxa414c27d93d5d5d5e72becc146c5f97b18
#> #define _prod _rxa414c27d93d5d5d5e72becc146c5f97b19
#> #define _max _rxa414c27d93d5d5d5e72becc146c5f97b20
#> #define _min _rxa414c27d93d5d5d5e72becc146c5f97b21
#> #define _transit4P _rxa414c27d93d5d5d5e72becc146c5f97b22
#> #define _transit3P _rxa414c27d93d5d5d5e72becc146c5f97b23
#> #define _assignFuns0 _rxa414c27d93d5d5d5e72becc146c5f97b24
#> #define _assignFuns _rxa414c27d93d5d5d5e72becc146c5f97b25
#> #define _rxord _rxa414c27d93d5d5d5e72becc146c5f97b26
#> #define __assignFuns2 _rxa414c27d93d5d5d5e72becc146c5f97b27
#> #define _llikCauchyDscale _rxa414c27d93d5d5d5e72becc146c5f97b28
#> #define _llikCauchyDlocation _rxa414c27d93d5d5d5e72becc146c5f97b29
#> #define _llikCauchy _rxa414c27d93d5d5d5e72becc146c5f97b30
#> #define _llikGammaDrate _rxa414c27d93d5d5d5e72becc146c5f97b31
#> #define _llikGammaDshape _rxa414c27d93d5d5d5e72becc146c5f97b32
#> #define _llikGamma _rxa414c27d93d5d5d5e72becc146c5f97b33
#> #define _llikWeibullDscale _rxa414c27d93d5d5d5e72becc146c5f97b34
#> #define _llikWeibullDshape _rxa414c27d93d5d5d5e72becc146c5f97b35
#> #define _llikWeibull _rxa414c27d93d5d5d5e72becc146c5f97b36
#> #define _llikUnifDbeta _rxa414c27d93d5d5d5e72becc146c5f97b37
#> #define _llikUnifDalpha _rxa414c27d93d5d5d5e72becc146c5f97b38
#> #define _llikUnif _rxa414c27d93d5d5d5e72becc146c5f97b39
#> #define _llikGeomDp _rxa414c27d93d5d5d5e72becc146c5f97b40
#> #define _llikGeom _rxa414c27d93d5d5d5e72becc146c5f97b41
#> #define _llikFDdf2 _rxa414c27d93d5d5d5e72becc146c5f97b42
#> #define _llikFDdf1 _rxa414c27d93d5d5d5e72becc146c5f97b43
#> #define _llikF _rxa414c27d93d5d5d5e72becc146c5f97b44
#> #define _llikExpDrate _rxa414c27d93d5d5d5e72becc146c5f97b45
#> #define _llikExp _rxa414c27d93d5d5d5e72becc146c5f97b46
#> #define _llikChisqDdf _rxa414c27d93d5d5d5e72becc146c5f97b47
#> #define _llikChisq _rxa414c27d93d5d5d5e72becc146c5f97b48
#> #define _llikTDsd _rxa414c27d93d5d5d5e72becc146c5f97b49
#> #define _llikTDmean _rxa414c27d93d5d5d5e72becc146c5f97b50
#> #define _llikTDdf _rxa414c27d93d5d5d5e72becc146c5f97b51
#> #define _llikT _rxa414c27d93d5d5d5e72becc146c5f97b52
#> #define _llikBetaDshape2 _rxa414c27d93d5d5d5e72becc146c5f97b53
#> #define _llikBetaDshape1 _rxa414c27d93d5d5d5e72becc146c5f97b54
#> #define _llikBeta _rxa414c27d93d5d5d5e72becc146c5f97b55
#> #define _llikNbinomMuDmu _rxa414c27d93d5d5d5e72becc146c5f97b56
#> #define _llikNbinomMu _rxa414c27d93d5d5d5e72becc146c5f97b57
#> #define _llikNbinomDprob _rxa414c27d93d5d5d5e72becc146c5f97b58
#> #define _llikNbinom _rxa414c27d93d5d5d5e72becc146c5f97b59
#> #define _llikBinomDprob _rxa414c27d93d5d5d5e72becc146c5f97b60
#> #define _llikBinom _rxa414c27d93d5d5d5e72becc146c5f97b61
#> #define _llikPoisDlambda _rxa414c27d93d5d5d5e72becc146c5f97b62
#> #define _llikPois _rxa414c27d93d5d5d5e72becc146c5f97b63
#> #define _llikNormDsd _rxa414c27d93d5d5d5e72becc146c5f97b64
#> #define _llikNormDmean _rxa414c27d93d5d5d5e72becc146c5f97b65
#> #define _llikNorm _rxa414c27d93d5d5d5e72becc146c5f97b66
#> #define simeps _rxa414c27d93d5d5d5e72becc146c5f97b67
#> #define simeta _rxa414c27d93d5d5d5e72becc146c5f97b68
#> #define expit _rxa414c27d93d5d5d5e72becc146c5f97b69
#> #define logit _rxa414c27d93d5d5d5e72becc146c5f97b70
#> #define gammapDer _rxa414c27d93d5d5d5e72becc146c5f97b71
#> #define lowergamma _rxa414c27d93d5d5d5e72becc146c5f97b72
#> #define uppergamma _rxa414c27d93d5d5d5e72becc146c5f97b73
#> #define gammaqInva _rxa414c27d93d5d5d5e72becc146c5f97b74
#> #define gammaqInv _rxa414c27d93d5d5d5e72becc146c5f97b75
#> #define gammapInva _rxa414c27d93d5d5d5e72becc146c5f97b76
#> #define gammapInv _rxa414c27d93d5d5d5e72becc146c5f97b77
#> #define gammaq _rxa414c27d93d5d5d5e72becc146c5f97b78
#> #define gammap _rxa414c27d93d5d5d5e72becc146c5f97b79
#> #define phi _rxa414c27d93d5d5d5e72becc146c5f97b80
#> #define riweibull _rxa414c27d93d5d5d5e72becc146c5f97b81
#> #define riunif _rxa414c27d93d5d5d5e72becc146c5f97b82
#> #define rit_ _rxa414c27d93d5d5d5e72becc146c5f97b83
#> #define ripois _rxa414c27d93d5d5d5e72becc146c5f97b84
#> #define ribeta _rxa414c27d93d5d5d5e72becc146c5f97b85
#> #define rigamma _rxa414c27d93d5d5d5e72becc146c5f97b86
#> #define rigeom _rxa414c27d93d5d5d5e72becc146c5f97b87
#> #define rif _rxa414c27d93d5d5d5e72becc146c5f97b88
#> #define riexp _rxa414c27d93d5d5d5e72becc146c5f97b89
#> #define richisq _rxa414c27d93d5d5d5e72becc146c5f97b90
#> #define ricauchy _rxa414c27d93d5d5d5e72becc146c5f97b91
#> #define rinbinomMu _rxa414c27d93d5d5d5e72becc146c5f97b92
#> #define rinbinom _rxa414c27d93d5d5d5e72becc146c5f97b93
#> #define ribinom _rxa414c27d93d5d5d5e72becc146c5f97b94
#> #define rinorm _rxa414c27d93d5d5d5e72becc146c5f97b95
#> #define rxweibull _rxa414c27d93d5d5d5e72becc146c5f97b96
#> #define rxunif _rxa414c27d93d5d5d5e72becc146c5f97b97
#> #define rxt_ _rxa414c27d93d5d5d5e72becc146c5f97b98
#> #define rxpois _rxa414c27d93d5d5d5e72becc146c5f97b99
#> #define rxbeta _rxa414c27d93d5d5d5e72becc146c5f97b100
#> #define rxgamma _rxa414c27d93d5d5d5e72becc146c5f97b101
#> #define rxgeom _rxa414c27d93d5d5d5e72becc146c5f97b102
#> #define rxf _rxa414c27d93d5d5d5e72becc146c5f97b103
#> #define rxexp _rxa414c27d93d5d5d5e72becc146c5f97b104
#> #define rxchisq _rxa414c27d93d5d5d5e72becc146c5f97b105
#> #define rxcauchy _rxa414c27d93d5d5d5e72becc146c5f97b106
#> #define rxnbinomMu _rxa414c27d93d5d5d5e72becc146c5f97b107
#> #define rxnbinom _rxa414c27d93d5d5d5e72becc146c5f97b108
#> #define rxbinom _rxa414c27d93d5d5d5e72becc146c5f97b109
#> #define rxnorm _rxa414c27d93d5d5d5e72becc146c5f97b110
#> #define linCmtC _rxa414c27d93d5d5d5e72becc146c5f97b111
#> #define linCmtB _rxa414c27d93d5d5d5e72becc146c5f97b112
#> #define linCmtA _rxa414c27d93d5d5d5e72becc146c5f97b113
#> #include <rxode2_model_shared.h>
#> #define __MAX_PROD__ 0
#> #define _CMT CMT
#> #define _SYNC_simeps_ for (int _svari=_solveData->neps; _svari--;){  if (_solveData->svar[_svari] == 0) {_THETA_1_ = _PP[0];};   if (_solveData->svar[_svari] == 1) {_THETA_2_ = _PP[1];};   if (_solveData->svar[_svari] == 2) {_THETA_3_ = _PP[2];};   if (_solveData->svar[_svari] == 3) {_THETA_4_ = _PP[3];};   if (_solveData->svar[_svari] == 4) {_THETA_5_ = _PP[4];};   if (_solveData->svar[_svari] == 5) {_ETA_1_ = _PP[5];};   if (_solveData->svar[_svari] == 6) {_ETA_2_ = _PP[6];};   if (_solveData->svar[_svari] == 7) {_ETA_3_ = _PP[7];}; }
#> #define _SYNC_simeta_ for (int _ovari=_solveData->neta; _ovari--;){  if (_solveData->ovar[_ovari] == 0) {_THETA_1_ = _PP[0];};   if (_solveData->ovar[_ovari] == 1) {_THETA_2_ = _PP[1];};   if (_solveData->ovar[_ovari] == 2) {_THETA_3_ = _PP[2];};   if (_solveData->ovar[_ovari] == 3) {_THETA_4_ = _PP[3];};   if (_solveData->ovar[_ovari] == 4) {_THETA_5_ = _PP[4];};   if (_solveData->ovar[_ovari] == 5) {_ETA_1_ = _PP[5];};   if (_solveData->ovar[_ovari] == 6) {_ETA_2_ = _PP[6];};   if (_solveData->ovar[_ovari] == 7) {_ETA_3_ = _PP[7];}; }
#> #include "extraC.h"
#> _getRxSolve_t _getRxSolve_;
#> _simfun simeps;
#> _simfun simeta;
#> rx_solve *_solveData=NULL;
#> rxode2_assign_ptr _assign_ptr=NULL;
#> _rxRmModelLibType _rxRmModelLib=NULL;
#> _rxGetModelLibType _rxGetModelLib=NULL;
#> rxode2_ode_solver_old_c _old_c=NULL;
#> rxode2_fn0i _ptrid=NULL;
#> _rxIsCurrentC_type _rxIsCurrentC=NULL;
#> _rxSumType _sumPS=NULL;
#> _rxProdType _prodPS=NULL;
#> rxode2_fn0i _prodType=NULL;
#> rxode2_fn0i _sumType=NULL;
#> _update_par_ptr_p _update_par_ptr=NULL;
#> _getParCov_p _getParCov=NULL;
#> linCmtA_p linCmtA;
#> linCmtA_p linCmtC;
#> linCmtB_p linCmtB;
#> _rx_asgn _rxode2_rxAssignPtr=NULL;
#> _rx_asgn _rxQr=NULL;
#> rxode2_fn phi;
#> rxode2_fn3 logit;
#> rxode2_fn3 expit;
#> rxode2_fn2 gammap;
#> rxode2_fn2 gammaq;
#> rxode2_fn2 lowergamma;
#> rxode2_fn2 uppergamma;
#> rxode2_fn2 gammapInv;
#> rxode2_fn2 gammapDer;
#> rxode2_fn2 gammapInva;
#> rxode2_fn2 gammaqInv;
#> rxode2_fn2 gammaqInva;
#> rxode2i_fn2 rxnorm;
#> rxode2i_rxbinom rxbinom;
#> rxode2i_rxbinom rxnbinom;
#> rxode2i_rxbinom rxnbinomMu;
#> rxode2i_fn2 rxcauchy;
#> rxode2i_fn rxchisq;
#> rxode2i_fn rxexp;
#> rxode2i_fn2 rxf;
#> rxode2i_ifn rxgeom;
#> rxode2i_fn2 rxgamma;
#> rxode2i_fn2 rxbeta;
#> rxode2i_ifn rxpois;
#> rxode2i_fn rxt_;
#> rxode2i_fn2 rxunif;
#> rxode2i_fn2 rxweibull;
#> rxode2i2_fn2 rinorm;
#> rxode2i2_ribinom ribinom;
#> rxode2i2_ribinom rinbinom;
#> rxode2i2_ribinom rinbinomMu;
#> rxode2i2_fn2 ricauchy;
#> rxode2i2_fn richisq;
#> rxode2i2_fn riexp;
#> rxode2i2_fn2 rif;
#> rxode2i2_ifn rigeom;
#> rxode2i2_fn2 rigamma;
#> rxode2i2_fn2 ribeta;
#> rxode2i2_ifn ripois;
#> rxode2i2_fn rit_;
#> rxode2i2_fn2 riunif;
#> rxode2i2_fn2 riweibull;
#> rxode2_llikNormFun _llikNorm;
#> rxode2_llikNormFun _llikNormDmean;
#> rxode2_llikNormFun _llikNormDsd;
#> rxode2_llikPoisFun _llikPois;
#> rxode2_llikPoisFun _llikPoisDlambda;
#> rxode2_llikBinomFun _llikBinom;
#> rxode2_llikBinomFun _llikBinomDprob;
#> rxode2_llikBinomFun _llikNbinom;
#> rxode2_llikBinomFun _llikNbinomDprob;
#> rxode2_llikBinomFun _llikNbinomMu;
#> rxode2_llikBinomFun _llikNbinomMuDmu;
#> rxode2_llikBetaFun _llikBeta;
#> rxode2_llikBetaFun _llikBetaDshape1;
#> rxode2_llikBetaFun _llikBetaDshape2;
#> rxode2_llikTFun _llikT;
#> rxode2_llikTFun _llikTDdf;
#> rxode2_llikTFun _llikTDmean;
#> rxode2_llikTFun _llikTDsd;
#> rxode2_llikChisqFun _llikChisq;
#> rxode2_llikChisqFun _llikChisqDdf;
#> rxode2_llikExpFun _llikExp;
#> rxode2_llikExpFun _llikExpDrate;
#> rxode2_llikFFun _llikF;
#> rxode2_llikFFun _llikFDdf1;
#> rxode2_llikFFun _llikFDdf2;
#> rxode2_llikGeomFun _llikGeom;
#> rxode2_llikGeomFun _llikGeomDp;
#> rxode2_llikUnifFun _llikUnif;
#> rxode2_llikUnifFun _llikUnifDalpha;
#> rxode2_llikUnifFun _llikUnifDbeta;
#> rxode2_llikWeibullFun _llikWeibull;
#> rxode2_llikWeibullFun _llikWeibullDshape;
#> rxode2_llikWeibullFun _llikWeibullDscale;
#> rxode2_llikGammaFun _llikGamma;
#> rxode2_llikGammaFun _llikGammaDshape;
#> rxode2_llikGammaFun _llikGammaDrate;
#> rxode2_llikCauchyFun _llikCauchy;
#> rxode2_llikCauchyFun _llikCauchyDlocation;
#> rxode2_llikCauchyFun _llikCauchyDscale;
#> rxode2_compareFactorVal_fn _compareFactorVal;
#> double _prod(double *input, double *p, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   return _prodPS(input, p, n, type);
#> }
#> double _sum(double *input, double *pld, int m, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   double ret = _sumPS(input, n, pld, m, type);
#>   if (type == 2 && m < 0){
#>     for (int i = -m; i--;){
#>       pld[i] = 0.0;
#>     }
#>   }
#>   return ret;
#> }
#> double _sign(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double s = 1;
#>   for (unsigned int i = 0; i < n; i++) {
#>     s = sign(va_arg(valist, double))*s;
#>     if (s == 0){
#>       break;
#>     }
#>   }
#>   va_end(valist);
#>   return s;
#> }
#> double _rxord(int _cSub, unsigned int n,  ...) {
#>   rx_solving_options_ind* ind = &(_solveData->subjects[_cSub]);
#>   if (!ind->inLhs) {
#>     return 1.0;
#>   }
#>   va_list valist;
#>   va_start(valist, n);
#>   double ret = 1.0;
#>   double p = 0.0;
#>   double u = rxunif(ind, 0.0, 1.0);
#>   int found = 0;
#>   for (unsigned int i = 0; i < n; i++) {
#>     p += va_arg(valist, double);
#>     if (!found) {
#>       if (u < p) {
#>         ret = (double)(i+1);
#>         found = 1;
#>       }
#>     }
#>   }
#>   if (!found) ret =(double)(n+1);
#>   va_end(valist);
#>   return ret;
#> }
#> double _max(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double mx = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mx = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++) {
#>       tmp = va_arg(valist, double);
#>       if (tmp>mx) mx=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mx;
#> }
#> double _min(unsigned int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   double mn = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mn = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++){
#>       tmp = va_arg(valist, double);
#>       if (tmp<mn) mn=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mn;
#> }
#> double _transit4P(int cmt, double t, unsigned int id, double n, double mtt, double bio){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = (t-tlast);
#>   return exp(_safe_log(bio*(_solveData->subjects[id].curDoseS[cmt]))+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> double _transit3P(int cmt, double t, unsigned int id, double n, double mtt){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = t-tlast;
#>   double podo = _solveData->subjects[id].curDoseS[cmt];
#>   return exp(_safe_log(podo)+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> void _assignFuns0(void) {
#>   _getRxSolve_ = (_getRxSolve_t) R_GetCCallable("rxode2","getRxSolve_");
#>   _assign_ptr=(rxode2_assign_ptr) R_GetCCallable("rxode2","rxode2_assign_fn_pointers");
#>   _rxRmModelLib=(_rxRmModelLibType) R_GetCCallable("rxode2","rxRmModelLib");
#>   _rxGetModelLib=(_rxGetModelLibType) R_GetCCallable("rxode2","rxGetModelLib");
#>   _rxode2_rxAssignPtr=(_rx_asgn)R_GetCCallable("rxode2","_rxode2_rxAssignPtr");
#>   _rxQr=(_rx_asgn)R_GetCCallable("rxode2parse","_rxode2parse_rxQr");
#>   _rxIsCurrentC = (_rxIsCurrentC_type)R_GetCCallable("rxode2","rxIsCurrentC");
#>   _sumPS  = (_rxSumType) R_GetCCallable("PreciseSums","PreciseSums_sum_r");
#>   _prodPS = (_rxProdType) R_GetCCallable("PreciseSums","PreciseSums_prod_r");
#>   _prodType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_prod_get");
#>   _sumType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_sum_get");
#>   _ptrid=(rxode2_fn0i)R_GetCCallable("rxode2", "rxode2_current_fn_pointer_id");
#>   _compareFactorVal=(rxode2_compareFactorVal_fn) R_GetCCallable("rxode2", "compareFactorVal");
#>   _update_par_ptr = (_update_par_ptr_p) R_GetCCallable("rxode2","_update_par_ptr");
#>   _getParCov = (_getParCov_p) R_GetCCallable("rxode2","_getParCov");
#>   // dynamic start
#>   _llikCauchyDscale = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDscale");
#>   _llikCauchyDlocation = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDlocation");
#>   _llikCauchy = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchy");
#>   _llikGammaDrate = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDrate");
#>   _llikGammaDshape = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDshape");
#>   _llikGamma = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGamma");
#>   _llikWeibullDscale = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDscale");
#>   _llikWeibullDshape = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDshape");
#>   _llikWeibull = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibull");
#>   _llikUnifDbeta = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDbeta");
#>   _llikUnifDalpha = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDalpha");
#>   _llikUnif = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnif");
#>   _llikGeomDp = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeomDp");
#>   _llikGeom = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeom");
#>   _llikFDdf2 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf2");
#>   _llikFDdf1 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf1");
#>   _llikF = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikF");
#>   _llikExpDrate = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExpDrate");
#>   _llikExp = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExp");
#>   _llikChisqDdf = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisqDdf");
#>   _llikChisq = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisq");
#>   _llikTDsd = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDsd");
#>   _llikTDmean = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDmean");
#>   _llikTDdf = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDdf");
#>   _llikT = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikT");
#>   _llikBetaDshape2 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape2");
#>   _llikBetaDshape1 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape1");
#>   _llikBeta = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBeta");
#>   _llikNbinomMuDmu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMuDmu");
#>   _llikNbinomMu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMu");
#>   _llikNbinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomDprob");
#>   _llikNbinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinom");
#>   _llikBinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinomDprob");
#>   _llikBinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinom");
#>   _llikPoisDlambda = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPoisDlambda");
#>   _llikPois = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPois");
#>   _llikNormDsd = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDsd");
#>   _llikNormDmean = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDmean");
#>   _llikNorm = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNorm");
#>   simeps = (_simfun) R_GetCCallable("rxode2random", "simeps");
#>   simeta = (_simfun) R_GetCCallable("rxode2random", "simeta");
#>   expit = (rxode2_fn3) R_GetCCallable("rxode2", "expit");
#>   logit = (rxode2_fn3) R_GetCCallable("rxode2", "logit");
#>   gammapDer = (rxode2_fn2) R_GetCCallable("rxode2", "gammapDer");
#>   lowergamma = (rxode2_fn2) R_GetCCallable("rxode2", "lowergamma");
#>   uppergamma = (rxode2_fn2) R_GetCCallable("rxode2", "uppergamma");
#>   gammaqInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInva");
#>   gammaqInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInv");
#>   gammapInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInva");
#>   gammapInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInv");
#>   gammaq = (rxode2_fn2) R_GetCCallable("rxode2", "gammaq");
#>   gammap = (rxode2_fn2) R_GetCCallable("rxode2", "gammap");
#>   phi = (rxode2_fn) R_GetCCallable("rxode2random", "phi");
#>   riweibull = (rxode2i2_fn2) R_GetCCallable("rxode2random", "riweibull");
#>   riunif = (rxode2i2_fn2) R_GetCCallable("rxode2random", "riunif");
#>   rit_ = (rxode2i2_fn) R_GetCCallable("rxode2random", "rit_");
#>   ripois = (rxode2i2_ifn) R_GetCCallable("rxode2random", "ripois");
#>   ribeta = (rxode2i2_fn2) R_GetCCallable("rxode2random", "ribeta");
#>   rigamma = (rxode2i2_fn2) R_GetCCallable("rxode2random", "rigamma");
#>   rigeom = (rxode2i2_ifn) R_GetCCallable("rxode2random", "rigeom");
#>   rif = (rxode2i2_fn2) R_GetCCallable("rxode2random", "rif");
#>   riexp = (rxode2i2_fn) R_GetCCallable("rxode2random", "riexp");
#>   richisq = (rxode2i2_fn) R_GetCCallable("rxode2random", "richisq");
#>   ricauchy = (rxode2i2_fn2) R_GetCCallable("rxode2random", "ricauchy");
#>   rinbinomMu = (rxode2i2_ribinom) R_GetCCallable("rxode2random", "rinbinomMu");
#>   rinbinom = (rxode2i2_ribinom) R_GetCCallable("rxode2random", "rinbinom");
#>   ribinom = (rxode2i2_ribinom) R_GetCCallable("rxode2random", "ribinom");
#>   rinorm = (rxode2i2_fn2) R_GetCCallable("rxode2random", "rinorm");
#>   rxweibull = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxweibull");
#>   rxunif = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxunif");
#>   rxt_ = (rxode2i_fn) R_GetCCallable("rxode2random", "rxt_");
#>   rxpois = (rxode2i_ifn) R_GetCCallable("rxode2random", "rxpois");
#>   rxbeta = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxbeta");
#>   rxgamma = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxgamma");
#>   rxgeom = (rxode2i_ifn) R_GetCCallable("rxode2random", "rxgeom");
#>   rxf = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxf");
#>   rxexp = (rxode2i_fn) R_GetCCallable("rxode2random", "rxexp");
#>   rxchisq = (rxode2i_fn) R_GetCCallable("rxode2random", "rxchisq");
#>   rxcauchy = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxcauchy");
#>   rxnbinomMu = (rxode2i_rxbinom) R_GetCCallable("rxode2random", "rxnbinomMu");
#>   rxnbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2random", "rxnbinom");
#>   rxbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2random", "rxbinom");
#>   rxnorm = (rxode2i_fn2) R_GetCCallable("rxode2random", "rxnorm");
#>   linCmtC = (linCmtA_p) R_GetCCallable("rxode2parse", "linCmtC");
#>   linCmtB = (linCmtB_p) R_GetCCallable("rxode2parse", "linCmtB");
#>   linCmtA = (linCmtA_p) R_GetCCallable("rxode2parse", "linCmtA");
#>   // dynamic stop
#>   _solveData = _getRxSolve_();
#> }
#> void _assignFuns(void) {
#>   if (_assign_ptr == NULL){
#>     _assignFuns0();
#>   }
#> }
#> void __assignFuns2(rx_solve rx,
#>                    rx_solving_options op,
#>                    t_F f,
#>                    t_LAG lag,
#>                    t_RATE rate,
#>                    t_DUR dur,
#>                    t_calc_mtime mtime,
#>                    t_ME me,
#>                    t_IndF indf,
#>                    t_getTime gettime,
#>                    t_locateTimeIndex timeindex,
#>                    t_handle_evidL handleEvid,
#>                    t_getDur getdur) {
#>   // assign start
#>     static rxode2_assignFuns2 rxode2parse_assignFuns2 = NULL;
#>     if (rxode2parse_assignFuns2 == NULL) rxode2parse_assignFuns2 = (rxode2_assignFuns2)(R_GetCCallable("rxode2parse", "_rxode2parse_assignFuns2"));
#>     rxode2parse_assignFuns2(rx, op, f, lag, rate, dur, mtime, me, indf, gettime, timeindex, handleEvid, getdur);
#>   // assign stop
#> }
#> extern void  rx_90b68597570124745132ef0971cce571_x64_ode_solver_solvedata (rx_solve *solve){
#>   _solveData = solve;
#> }
#> extern rx_solve *rx_90b68597570124745132ef0971cce571_x64_ode_solver_get_solvedata(void){
#>   return _solveData;
#> }
#> SEXP rx_90b68597570124745132ef0971cce571_x64_model_vars(void);
#> 
#> 
#> // prj-specific differential eqns
#> void rx_90b68597570124745132ef0971cce571_x64_dydt(int *_neq, double __t, double *__zzStateVar__, double *__DDtStateVar__)
#> {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   int _cSub = _neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=1;
#>     double _THETA_1_;
#>   double _THETA_2_;
#>   double _THETA_3_;
#>   double _THETA_4_;
#>   double _THETA_5_;
#>   double _ETA_1_;
#>   double _ETA_2_;
#>   double _ETA_3_;
#>   double depot;
#>   double center;
#>   double rx_expr_0;
#>   double rx_expr_3;
#>   double rx_expr_1;
#>   double rx_expr_2;
#>   double rx_yj___;
#>   double rx_lambda___;
#>   double rx_hi___;
#>   double rx_low___;
#>   double rx_expr_4;
#>   double rx_expr_5;
#>   double rx_pred_;
#>   double rx_r_;
#>   double tka;
#>   double tcl;
#>   double tv;
#>   double add_DoT_sd;
#>   double pow;
#>   double eta_DoT_ka;
#>   double eta_DoT_cl;
#>   double eta_DoT_v;
#>   double ka;
#>   double cl;
#>   double v;
#>   double cp;
#>   double tad;
#>   double dosenum;
#> 
#>   (void)t;
#>   (void)_THETA_1_;
#>   (void)_THETA_2_;
#>   (void)_THETA_3_;
#>   (void)_THETA_4_;
#>   (void)_THETA_5_;
#>   (void)_ETA_1_;
#>   (void)_ETA_2_;
#>   (void)_ETA_3_;
#>   (void)depot;
#>   (void)center;
#>   (void)rx_expr_0;
#>   (void)rx_expr_3;
#>   (void)rx_expr_1;
#>   (void)rx_expr_2;
#>   (void)rx_yj___;
#>   (void)rx_lambda___;
#>   (void)rx_hi___;
#>   (void)rx_low___;
#>   (void)rx_expr_4;
#>   (void)rx_expr_5;
#>   (void)rx_pred_;
#>   (void)rx_r_;
#>   (void)tka;
#>   (void)tcl;
#>   (void)tv;
#>   (void)add_DoT_sd;
#>   (void)pow;
#>   (void)eta_DoT_ka;
#>   (void)eta_DoT_cl;
#>   (void)eta_DoT_v;
#>   (void)ka;
#>   (void)cl;
#>   (void)v;
#>   (void)cp;
#>   (void)tad;
#>   (void)dosenum;
#> 
#>   rx_pred_ = _PL[0];
#>   rx_r_ = _PL[1];
#>   tka = _PL[2];
#>   tcl = _PL[3];
#>   tv = _PL[4];
#>   add_DoT_sd = _PL[5];
#>   pow = _PL[6];
#>   eta_DoT_ka = _PL[7];
#>   eta_DoT_cl = _PL[8];
#>   eta_DoT_v = _PL[9];
#>   ka = _PL[10];
#>   cl = _PL[11];
#>   v = _PL[12];
#>   cp = _PL[13];
#>   tad = _PL[14];
#>   dosenum = _PL[15];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   _THETA_1_ = _PP[0];
#>   _THETA_2_ = _PP[1];
#>   _THETA_3_ = _PP[2];
#>   _THETA_4_ = _PP[3];
#>   _THETA_5_ = _PP[4];
#>   _ETA_1_ = _PP[5];
#>   _ETA_2_ = _PP[6];
#>   _ETA_3_ = _PP[7];
#> 
#>   depot = __zzStateVar__[0]*((double)(_ON[0]));
#>   center = __zzStateVar__[1]*((double)(_ON[1]));
#> 
#>   rx_expr_0 =_ETA_1_+_THETA_1_;
#>   rx_expr_3 =exp(rx_expr_0);
#>   __DDtStateVar__[0] = ((double)(_ON[0]))*(_IR[0] -rx_expr_3*depot);
#>   rx_expr_1 =_ETA_2_+_THETA_2_;
#>   rx_expr_2 =_ETA_3_+_THETA_3_;
#>   __DDtStateVar__[1] = ((double)(_ON[1]))*(_IR[1] + rx_expr_3*depot-exp(rx_expr_1-(rx_expr_2))*center);
#>   rx_yj_ =2;
#>   rx_lambda_ =1;
#>   rx_hi_ =1;
#>   rx_low_ =0;
#>   rx_expr_4 =exp(-(rx_expr_2));
#>   rx_expr_5 =rx_expr_4*center;
#>   rx_pred_=rx_expr_5;
#>   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>   tka=_THETA_1_;
#>   tcl=_THETA_2_;
#>   tv=_THETA_3_;
#>   add_DoT_sd=_THETA_4_;
#>   pow=_THETA_5_;
#>   eta_DoT_ka=_ETA_1_;
#>   eta_DoT_cl=_ETA_2_;
#>   eta_DoT_v=_ETA_3_;
#>   ka=rx_expr_3;
#>   cl=exp(rx_expr_1);
#>   v=exp(rx_expr_2);
#>   cp=rx_expr_5;
#>   tad=_tad0();
#>   dosenum=(double)(_solveData->subjects[_cSub].dosenum);
#>   (&_solveData->subjects[_cSub])->dadt_counter[0]++;
#> }
#> 
#> // Jacobian derived vars
#> void rx_90b68597570124745132ef0971cce571_x64_calc_jac(int *_neq, double __t, double *__zzStateVar__, double *__PDStateVar__, unsigned int __NROWPD__) {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>     int _cSub=_neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=2;
#>     (&_solveData->subjects[_cSub])->jac_counter[0]++;
#> }
#> // Functional based initial conditions.
#> void rx_90b68597570124745132ef0971cce571_x64_inis(int _cSub, double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   
#>   (&_solveData->subjects[_cSub])->_rxFlag=3;
#>   }
#> // prj-specific derived vars
#> void rx_90b68597570124745132ef0971cce571_x64_calc_lhs(int _cSub, double __t, double *__zzStateVar__, double *_lhs) {
#>     int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=11;
#>     double  __DDtStateVar_0__;
#>   double  __DDtStateVar_1__;
#>   double  __DDtStateVar_2__;
#>   double _THETA_1_;
#>   double _THETA_2_;
#>   double _THETA_3_;
#>   double _THETA_4_;
#>   double _THETA_5_;
#>   double _ETA_1_;
#>   double _ETA_2_;
#>   double _ETA_3_;
#>   double depot;
#>   double center;
#>   double rx_expr_0;
#>   double rx_expr_3;
#>   double rx_expr_1;
#>   double rx_expr_2;
#>   double rx_yj___;
#>   double rx_lambda___;
#>   double rx_hi___;
#>   double rx_low___;
#>   double rx_expr_4;
#>   double rx_expr_5;
#>   double rx_pred_;
#>   double rx_r_;
#>   double tka;
#>   double tcl;
#>   double tv;
#>   double add_DoT_sd;
#>   double pow;
#>   double eta_DoT_ka;
#>   double eta_DoT_cl;
#>   double eta_DoT_v;
#>   double ka;
#>   double cl;
#>   double v;
#>   double cp;
#>   double tad;
#>   double dosenum;
#> 
#>   (void)t;
#>   (void)__DDtStateVar_0__;
#>   (void)__DDtStateVar_1__;
#>   (void)__DDtStateVar_2__;
#>   (void)_THETA_1_;
#>   (void)_THETA_2_;
#>   (void)_THETA_3_;
#>   (void)_THETA_4_;
#>   (void)_THETA_5_;
#>   (void)_ETA_1_;
#>   (void)_ETA_2_;
#>   (void)_ETA_3_;
#>   (void)depot;
#>   (void)center;
#>   (void)rx_expr_0;
#>   (void)rx_expr_3;
#>   (void)rx_expr_1;
#>   (void)rx_expr_2;
#>   (void)rx_yj___;
#>   (void)rx_lambda___;
#>   (void)rx_hi___;
#>   (void)rx_low___;
#>   (void)rx_expr_4;
#>   (void)rx_expr_5;
#>   (void)rx_pred_;
#>   (void)rx_r_;
#>   (void)tka;
#>   (void)tcl;
#>   (void)tv;
#>   (void)add_DoT_sd;
#>   (void)pow;
#>   (void)eta_DoT_ka;
#>   (void)eta_DoT_cl;
#>   (void)eta_DoT_v;
#>   (void)ka;
#>   (void)cl;
#>   (void)v;
#>   (void)cp;
#>   (void)tad;
#>   (void)dosenum;
#> 
#>   rx_pred_ = _PL[0];
#>   rx_r_ = _PL[1];
#>   tka = _PL[2];
#>   tcl = _PL[3];
#>   tv = _PL[4];
#>   add_DoT_sd = _PL[5];
#>   pow = _PL[6];
#>   eta_DoT_ka = _PL[7];
#>   eta_DoT_cl = _PL[8];
#>   eta_DoT_v = _PL[9];
#>   ka = _PL[10];
#>   cl = _PL[11];
#>   v = _PL[12];
#>   cp = _PL[13];
#>   tad = _PL[14];
#>   dosenum = _PL[15];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   _THETA_1_ = _PP[0];
#>   _THETA_2_ = _PP[1];
#>   _THETA_3_ = _PP[2];
#>   _THETA_4_ = _PP[3];
#>   _THETA_5_ = _PP[4];
#>   _ETA_1_ = _PP[5];
#>   _ETA_2_ = _PP[6];
#>   _ETA_3_ = _PP[7];
#> 
#>   depot = __zzStateVar__[0]*((double)(_ON[0]));
#>   center = __zzStateVar__[1]*((double)(_ON[1]));
#> 
#>   rx_expr_0 =_ETA_1_+_THETA_1_;
#>   rx_expr_3 =exp(rx_expr_0);
#>   __DDtStateVar_0__ = ((double)(_ON[0]))*(_IR[0] -rx_expr_3*depot);
#>   rx_expr_1 =_ETA_2_+_THETA_2_;
#>   rx_expr_2 =_ETA_3_+_THETA_3_;
#>   __DDtStateVar_1__ = ((double)(_ON[1]))*(_IR[1] + rx_expr_3*depot-exp(rx_expr_1-(rx_expr_2))*center);
#>   rx_yj_ =2;
#>   rx_lambda_ =1;
#>   rx_hi_ =1;
#>   rx_low_ =0;
#>   rx_expr_4 =exp(-(rx_expr_2));
#>   rx_expr_5 =rx_expr_4*center;
#>   rx_pred_=rx_expr_5;
#>   rx_r_=Rx_pow_di((Rx_pow(_as_dbleps((rx_expr_5)),_THETA_5_)*_THETA_4_),2);
#>   tka=_THETA_1_;
#>   tcl=_THETA_2_;
#>   tv=_THETA_3_;
#>   add_DoT_sd=_THETA_4_;
#>   pow=_THETA_5_;
#>   eta_DoT_ka=_ETA_1_;
#>   eta_DoT_cl=_ETA_2_;
#>   eta_DoT_v=_ETA_3_;
#>   ka=rx_expr_3;
#>   cl=exp(rx_expr_1);
#>   v=exp(rx_expr_2);
#>   cp=rx_expr_5;
#>   tad=_tad0();
#>   dosenum=(double)(_solveData->subjects[_cSub].dosenum);
#> 
#>   _lhs[0]=rx_pred_;
#>   _lhs[1]=rx_r_;
#>   _lhs[2]=tka;
#>   _lhs[3]=tcl;
#>   _lhs[4]=tv;
#>   _lhs[5]=add_DoT_sd;
#>   _lhs[6]=pow;
#>   _lhs[7]=eta_DoT_ka;
#>   _lhs[8]=eta_DoT_cl;
#>   _lhs[9]=eta_DoT_v;
#>   _lhs[10]=ka;
#>   _lhs[11]=cl;
#>   _lhs[12]=v;
#>   _lhs[13]=cp;
#>   _lhs[14]=tad;
#>   _lhs[15]=dosenum;
#> }
#> // Functional based bioavailability
#> double rx_90b68597570124745132ef0971cce571_x64_F(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>  return _amt;
#>   }
#> // Functional based absorption lag
#> double rx_90b68597570124745132ef0971cce571_x64_Lag(int _cSub,  int _cmt, double __t, double *__zzStateVar__){
#>  return __t;
#> }
#> // Modeled zero-order rate
#> double rx_90b68597570124745132ef0971cce571_x64_Rate(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>  return 0.0;
#> }
#> // Modeled zero-order duration
#> double rx_90b68597570124745132ef0971cce571_x64_Dur(int _cSub,  int _cmt, double _amt, double __t){
#>  return 0.0;
#> }
#> // Model Times
#> void rx_90b68597570124745132ef0971cce571_x64_mtime(int _cSub, double *_mtime){
#> }
#> // Matrix Exponential (0)
#> void rx_90b68597570124745132ef0971cce571_x64_ME(int _cSub, double _t, double __t, double *_mat, const double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=9;
#>   }
#> // Inductive linearization Matf
#> void rx_90b68597570124745132ef0971cce571_x64_IndF(int _cSub, double _t, double __t, double *_matf){
#>  int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=10;
#>   }
#> extern SEXP rx_90b68597570124745132ef0971cce571_x64_model_vars(void){
#>   int pro=0;
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_90b68597570124745132ef0971cce571_x64_model_vars"));pro++;
#>   if (!_rxIsCurrentC(_mv)){
#>     SEXP hash    = PROTECT(allocVector(STRSXP, 1));pro++;
#> #define __doBuf__  snprintf(buf, __doBufN__, "un]\"BAAA@QRtHACAAAAAAABt+AAAv7#aT)O4HrRAll^<>WrmYBdp+k1+B[6Cfo0wo~HSN]~BI!]ZT=@cy{!l+jxbNfni#&p...d&_]^iTw2u$kwYhmIl0`AS*8qLBvE\"Lt^AgYO4^5@9GO3|x:<!4e%%6e$$Mc^#ZD}Cn[55<e[D3EWQz8>dyozRT/#>4y0QHwjqErUC#mSFp$c)*w,BxOR8n=kOek)Yx7Rd9L,:yx%%L],#$tAcR@suX{.9Ys^A.N2Ej3]T+GXOV]$g:2w]&Cb`j12ry/X*<m|Jbb_,ciHN5SH0|!JE[kHeo=[P`zR)bSDvcSoY>JsBg<<woQiL,\?nf*QE`F0JIsMY%%[$21gj@V1]yJB}(ocT:Mk\?k,gPGM4BZK.fhx)P_}Jha.`QWK~wHD<1Mv/R@00o`<oda.ZEOv1)@g8pQAbA6y]aT<R*.OL0E0ALhCtIaL%%wVIsN=H`.o2Z;oSv8Z3H0UVs;UCEr+17*Xk+i+HWJtu5}A;%%b/]m.CN%%=ICE%%CMOil`coJ{IB/rpWW(>F2e~+P2y>SVt)kZJYSwAwlBAYD*rBBzEM,>V[iG9=Qd#t+qf~XppH:RAzk:8S>R[~S,jeE@i.p<#$.ff.&j:yVpcu\"`Kv|GVU2&oq@{l&{Gy$ZW81i\"+$%%&pzV^*,b!$uIY\?|W+bEpBwlEo.##2q\?iL:y0oE#Ue!y9];2G|wn+MO[A+Qa~nIYDKIT$nw|CaQ]:G%%<h.XubB2<0LGy1Bs^`zukG::8A6HoVyHpk,~#l[|y{<R9#`RCRaC938g5QLF7Eu^izVJ@Y~)wxOuSr[9wm}469V7w_uEn\?{2zx;[H$M)RHp9NF$0oJr|R1=d:WS5oL$@LltTj9XZq4oqj[&Zj;e_jPRVA@f]hO_w<s90!Anm9ZPMVYC!@^w:l\"f2e0l\"f=%%CP5h`&9)AA/wsqm0VXRG#u=M|W!kBRnupl8:,ye+a2|T8k+yK*)ML*TwkApm!O_y2!HNoFqz[F05&ia0.}kWSdGUDW$4I<6s@H(2V5Ke8/yjuLhfsJSbw3[Jn<xL<ruVg/x}cnab(rVxU}|AQs\"Fy4>.h<4mQuI.p1x1;yT4(3dRP>m}FlXy+~p:cCJ+3K6eyDhV@gEeudUfPP%%r01$K`300[KioZAo3HQk6*~)!6SeGF#V5#8l4I8VoSw\"`trSNvtHddH7c^DLHs4O9$Pd|yc:.[n=X[01[=x(|^]*ksuwhQq@Za(/3waRy=*a5NMQ3~Q}M`$9wHI8zDaQWM&M*c6dB}KI\?UWGSA{ETxN\"`]0QQyw3UtHUd]QT1FPYgGFK)s<_X$P>D{(Uvam\?k\?]uB!`(/v=u39lz;/}L5o/fID%%s\"0IwW[:6h=:Cw!FycuCn6*_(%%`+vspc=uXkg=%%LHiV8^Hi|\"HTo#6a}ahso%%rcU`+.B");
#>     char buf[1436];
#> #define __doBufN__ 1436
#>     __doBuf__
#> #undef __doBuf__
#> #undef __doBufN__
#>     SET_STRING_ELT(hash, 0, mkChar(buf));
#>     SEXP lst      = PROTECT(_rxQr(hash));pro++;
#>     _assign_ptr(lst);
#>     UNPROTECT(pro);
#>     return lst;
#>   } else {
#>     UNPROTECT(pro);
#>     return _mv;
#>   }
#> }
#> extern void rx_90b68597570124745132ef0971cce571_x64_dydt_lsoda(int *neq, double *t, double *A, double *DADT)
#> {
#>   rx_90b68597570124745132ef0971cce571_x64_dydt(neq, *t, A, DADT);
#> }
#> extern int rx_90b68597570124745132ef0971cce571_x64_dydt_liblsoda(double __t, double *y, double *ydot, void *data)
#> {
#>   int *neq = (int*)(data);
#>   rx_90b68597570124745132ef0971cce571_x64_dydt(neq, __t, y, ydot);
#>   return(0);
#> }
#> extern void rx_90b68597570124745132ef0971cce571_x64_calc_jac_lsoda(int *neq, double *t, double *A,int *ml, int *mu, double *JAC, int *nrowpd){
#>   // Update all covariate parameters
#>   rx_90b68597570124745132ef0971cce571_x64_calc_jac(neq, *t, A, JAC, *nrowpd);
#> }
#> 
#> //Create function to call from R's main thread that assigns the required functions. Sometimes they don't get assigned.
#> extern void rx_90b68597570124745132ef0971cce571_x64_assignFuns(void){
#>   _assignFuns();
#> }
#> 
#> //Initialize the dll to match rxode2's calls
#> void R_init0_rx_90b68597570124745132ef0971cce571_x64(void){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_assignFuns2", (DL_FUNC) __assignFuns2);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_assignFuns", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_assignFuns);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_inis",(DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_inis);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_dydt",(DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_dydt);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_calc_lhs",(DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_calc_lhs);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_calc_jac",(DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_calc_jac);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_dydt_lsoda", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_dydt_lsoda);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_calc_jac_lsoda", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_calc_jac_lsoda);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_ode_solver_solvedata", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_ode_solver_solvedata);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_ode_solver_get_solvedata", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_ode_solver_get_solvedata);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_F", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_F);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_Lag", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_Lag);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_Rate", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_Rate);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_Dur", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_Dur);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_mtime", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_mtime);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_ME", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_ME);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_IndF", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_IndF);
#>   R_RegisterCCallable("rx_90b68597570124745132ef0971cce571_x64","rx_90b68597570124745132ef0971cce571_x64_dydt_liblsoda", (DL_FUNC) rx_90b68597570124745132ef0971cce571_x64_dydt_liblsoda);
#> }
#> //Initialize the dll to match rxode2's calls
#> void R_init_rx_90b68597570124745132ef0971cce571_x64(DllInfo *info){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_init0_rx_90b68597570124745132ef0971cce571_x64();
#>   static const R_CallMethodDef callMethods[]  = {
#>     {"rx_90b68597570124745132ef0971cce571_x64_model_vars", (DL_FUNC) &rx_90b68597570124745132ef0971cce571_x64_model_vars, 0},
#>     {NULL, NULL, 0}
#>   };
#> 
#>   R_registerRoutines(info, NULL, callMethods, NULL, NULL);
#>   R_useDynamicSymbols(info,FALSE);
#>   _assignFuns0();
#> 
#> }
#> 
#> void R_unload_rx_90b68597570124745132ef0971cce571_x64 (DllInfo *info){
#>   // Free resources required for single subject solve.
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_90b68597570124745132ef0971cce571_x64_model_vars"));
#>   if (!isNull(_mv)){
#>     _rxRmModelLib("rx_90b68597570124745132ef0971cce571_x64_model_vars");
#>   }
#>   UNPROTECT(1);
#> }
#> Error : error building model
#> Error: error building model
```

<sup>Created on 2023-01-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

--- a/NEWS.md
+++ b/NEWS.md
@@ -7,6 +7,9 @@
 
 * Add check for correct number of function arguments to parser
 
+* Like R, known functions can be assigned as a variable and the
+  function can still be called (while not changing the variable value)
+
 * Fix garbled error messages that occur with certain messages.
 
 # rxode2parse 2.0.13

--- a/inst/include/sbuf.c
+++ b/inst/include/sbuf.c
@@ -219,7 +219,15 @@ void curLineType(vLines *sbb, int propId) {
   sbb->lType[sbb->n] = propId;
 }
 
+extern SEXP _goodFuns;
+
 void doDot(sbuf *out, char *buf) {
+  for (int j = Rf_length(_goodFuns); j--;){
+    if (!strcmp(CHAR(STRING_ELT(_goodFuns, j)), buf)) {
+      sAppendN(out, "_rxNotFun_", 10);
+      break;
+    }
+  }
   for (int k = 0; k < (int)strlen(buf); k++){
     if (buf[k] == '.'){
       sAppend(out,"_DoT_");
@@ -230,6 +238,13 @@ void doDot(sbuf *out, char *buf) {
 }
 
 void doDot2(sbuf *sb, sbuf *sbDt, char *buf) {
+  for (int j = Rf_length(_goodFuns); j--;){
+    if (!strcmp(CHAR(STRING_ELT(_goodFuns, j)), buf)) {
+      sAppendN(sb, "_rxNotFun_", 10);
+      sAppendN(sbDt, "_rxNotFun_", 10);
+      break;
+    }
+  }
   for (int k = 0; k < (int)strlen(buf); k++) {
     if (buf[k] == '.') {
       sAppend(sb,"_DoT_");
@@ -240,5 +255,3 @@ void doDot2(sbuf *sb, sbuf *sbDt, char *buf) {
     }
   }
 }
-
-

--- a/src/parseIdentifier.h
+++ b/src/parseIdentifier.h
@@ -8,23 +8,23 @@ static inline void handleIdentifier(nodeInfo ni, char *name, char *value) {
       addSymbolStr(value);
       // Ignored variables
       if (isTbsVar(value)){
-	// If it is Transform both sides, suppress printouts
-	tb.lh[NV-1] = isSuppressedParam; // Suppress param printout.
+        // If it is Transform both sides, suppress printouts
+        tb.lh[NV-1] = isSuppressedParam; // Suppress param printout.
       }
     } else if (isDefiningParameterRecursively(value)){
       // This is x = x*exp(matt)
       // lhs defined in terms of a parameter
       if (tb.lh[tb.ix] == isSuppressedLHS){
-	tb.lh[tb.ix] = notLHS;
+        tb.lh[tb.ix] = notLHS;
       } else {
-	tb.lh[tb.ix] = isLHSparam;
+        tb.lh[tb.ix] = isLHSparam;
       }
     }
   }
 }
 
 static inline void handleOperatorsOrPrintingIdentifiers(int depth, print_node_fn_t fn, void *client_data,
-							nodeInfo ni, char *name, char *value) {
+                                                        nodeInfo ni, char *name, char *value) {
   if (isOperatorOrPrintingIdentifier(ni, name))
     fn(depth, name, value, client_data);
   if (!strcmp("=", name)){
@@ -99,15 +99,15 @@ static inline void handleSafeZero(nodeInfo ni, char *name, int i, int *safe_zero
     char *v = (char*)rc_dup_str(xpn->start_loc.s, xpn->end);
     if (i == 0){
       if (!strcmp("/",v)){
-	aAppendN("safe_zero(", 10);
-	*safe_zero = 1;
+        aAppendN("safe_zero(", 10);
+        *safe_zero = 1;
       } else {
-	*safe_zero = 0;
+        *safe_zero = 0;
       }
     }
     if (i == 1){
       if (*safe_zero){
-	aAppendN(")", 1);
+        aAppendN(")", 1);
       }
       *safe_zero = 0;
     }

--- a/src/print_node.c
+++ b/src/print_node.c
@@ -3,6 +3,8 @@
 #include "print_node.h"
 #include "tran.h"
 
+extern SEXP _goodFuns;
+
 void wprint_node(int depth, char *name, char *value, void *client_data) {
   int i;
   nodeInfo ni;
@@ -25,6 +27,15 @@ void wprint_node(int depth, char *name, char *value, void *client_data) {
       nodeFunLinCmtC(value);
   }
   if (!tmp) {
+    // See if this variable is a function too
+    if (!nodeHas(identifier)) {
+      for (int j = Rf_length(_goodFuns); j--;){
+        if (!strcmp(CHAR(STRING_ELT(_goodFuns, j)),value)) {
+          aAppendN("_rxNotFun_", 10);
+          break;
+        }
+      }
+    }
     // Apply fix for dot.syntax
     for (i = 0; i < (int)strlen(value); i++){
       if (value[i] == '.' && nodeHas(identifier_r)){

--- a/tests/testthat/test-parsing.R
+++ b/tests/testthat/test-parsing.R
@@ -682,3 +682,20 @@ test_that("TIME conundrums", {
   expect_false(any(p$params == "TIME"))
   
 })
+
+
+test_that("pow problems", {
+  expect_error(rxode2parse("pow=3+4"), NA)
+
+  rxode2parse("pow = 3+1+pow(4, 3)\npow2 = pow*2", code="rxode2parse_test_code.c")
+
+  expect_true(file.exists("rxode2parse_test_code.c"))
+  if (file.exists("rxode2parse_test_code.c")) {
+    lines <- readLines("rxode2parse_test_code.c")
+    unlink("rxode2parse_test_code.c")
+    expect_false(file.exists("rxode2parse_test_code.c"))
+    expect_true(any(regexpr("+pow(4,3)", lines, fixed=TRUE) != -1))
+    expect_true(any(regexpr("_rxNotFun_pow=", lines, fixed=TRUE) != -1))
+    expect_true(any(regexpr("=_rxNotFun_pow*2", lines, fixed=TRUE) != -1))
+  }
+})

